### ISSUE TITLE
go cannot find main module?

### ISSUE THREAD

**Issue by dhammel** [hardcore-sushi/DroidFS#39]
Hello,

OS: Kubuntu 21.04
Go Version: 1.16.5
NDK Version: 21.4.7075529
Latest version of Android Studio Stable

when attempting to build DroidFS from any release tag of v1.4.5 & higher, then last command according to readme when being run take a few minutes and displays error:

make[1]: Leaving directory '/home/dhammel/Android/Project/DroidFS/app/libgocryptfs/openssl-1.1.1k'
go: cannot find main module, but found .git/config in /home/dhammel/Android/Project/DroidFS
        to create a module there, run:
        cd ../.. && go mod init
Done.

it seems as if the go build command, nearly the last step of the build.sh script, seem to cause this error and I can't seem to find out why. libgocrypfs.so, doesn't get built because of this.

I followed all directions of readme, so i have all the needed go modules, libgocrypfs submodule is included, openssl...

Build release tag v1.4.4 or below works, anything higher yields this error.

**Comment by hardcore-sushi** [comment]
Does it show the same error message when running manually `go build` inside `app/libgocryptfs` ?
If it does, try to run `go mod init libgocryptfs` then retry to run `build.sh`.

**Comment by dhammel** [comment]
When running go build in side app/libgocryptfs, it does show the same error. Running go mod init libgocryptfs was a success, but when running the build.sh script, it now gives a different error: 

make[1]: Leaving directory '/home/dhammel/Android/Project/DroidFS/app/libgocryptfs/openssl-1.1.1k'
directory.go:13:2: local import "./allocator" in non-local package
directory.go:14:2: local import "./internal/configfile" in non-local package
file.go:11:2: local import "./internal/contentenc" in non-local package
directory.go:15:2: local import "./internal/cryptocore" in non-local package
common_ops.go:9:2: local import "./internal/nametransform" in non-local package
volume.go:17:2: local import "./internal/stupidgcm" in non-local package
common_ops.go:10:2: local import "./internal/syscallcompat" in non-local package
internal/siv_aead/siv_aead.go:9:2: cannot find package
internal/cryptocore/cryptocore.go:13:2: cannot find package
internal/cryptocore/hkdf.go:7:2: cannot find package
internal/configfile/scrypt.go:8:2: cannot find package
internal/stupidgcm/prefer.go:6:2: cannot find package
internal/syscallcompat/eintr.go:6:2: no required module provides package golang.org/x/sys/unix; to add it:
        go get golang.org/x/sys/unix
Done.

seems as if certain modules aren't found, but i've ran the go get command according to the README, and they are stored in ~/go/pkg/

**Comment by hardcore-sushi** [comment]
OK, undo the `go mod init` (`rm go.mod go.sum`).
Show me the output of `go env`. If you see `GO111MODULE="on"`, run `unset GO111MODULE` or `go env -w GO111MODULE=auto` then try again.

**Comment by dhammel** [comment]
The go env:

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/dhammel/.cache/go-build"
GOENV="/home/dhammel/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/dhammel/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/dhammel/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/home/dhammel/Apps/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/home/dhammel/Apps/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.5"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build814030323=/tmp/go-build -gno-record-gcc-switches"

I ran switched GO111MODULE to auto, per your command, and the error in when running the build script is as follows:

Compiling for x86_64...
internal/siv_aead/siv_aead.go:9:2: cannot find package "github.com/jacobsa/crypto/siv" in any of:
        /home/dhammel/Apps/go/src/github.com/jacobsa/crypto/siv (from $GOROOT)
        /home/dhammel/go/src/github.com/jacobsa/crypto/siv (from $GOPATH)
internal/cryptocore/cryptocore.go:13:2: cannot find package "github.com/rfjakob/eme" in any of:
        /home/dhammel/Apps/go/src/github.com/rfjakob/eme (from $GOROOT)
        /home/dhammel/go/src/github.com/rfjakob/eme (from $GOPATH)
internal/cryptocore/hkdf.go:7:2: cannot find package "golang.org/x/crypto/hkdf" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/hkdf (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/hkdf (from $GOPATH)
internal/configfile/scrypt.go:8:2: cannot find package "golang.org/x/crypto/scrypt" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/scrypt (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/scrypt (from $GOPATH)
internal/stupidgcm/prefer.go:6:2: cannot find package "golang.org/x/sys/cpu" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/cpu (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/cpu (from $GOPATH)
internal/syscallcompat/eintr.go:6:2: cannot find package "golang.org/x/sys/unix" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/unix (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/unix (from $GOPATH)
Compiling for x86...
internal/siv_aead/siv_aead.go:9:2: cannot find package "github.com/jacobsa/crypto/siv" in any of:
        /home/dhammel/Apps/go/src/github.com/jacobsa/crypto/siv (from $GOROOT)
        /home/dhammel/go/src/github.com/jacobsa/crypto/siv (from $GOPATH)
internal/cryptocore/cryptocore.go:13:2: cannot find package "github.com/rfjakob/eme" in any of:
        /home/dhammel/Apps/go/src/github.com/rfjakob/eme (from $GOROOT)
        /home/dhammel/go/src/github.com/rfjakob/eme (from $GOPATH)
internal/cryptocore/hkdf.go:7:2: cannot find package "golang.org/x/crypto/hkdf" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/hkdf (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/hkdf (from $GOPATH)
internal/configfile/scrypt.go:8:2: cannot find package "golang.org/x/crypto/scrypt" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/scrypt (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/scrypt (from $GOPATH)
internal/stupidgcm/prefer.go:6:2: cannot find package "golang.org/x/sys/cpu" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/cpu (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/cpu (from $GOPATH)
internal/syscallcompat/eintr.go:6:2: cannot find package "golang.org/x/sys/unix" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/unix (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/unix (from $GOPATH)
Compiling for arm64-v8a...
internal/siv_aead/siv_aead.go:9:2: cannot find package "github.com/jacobsa/crypto/siv" in any of:
        /home/dhammel/Apps/go/src/github.com/jacobsa/crypto/siv (from $GOROOT)
        /home/dhammel/go/src/github.com/jacobsa/crypto/siv (from $GOPATH)
internal/cryptocore/cryptocore.go:13:2: cannot find package "github.com/rfjakob/eme" in any of:
        /home/dhammel/Apps/go/src/github.com/rfjakob/eme (from $GOROOT)
        /home/dhammel/go/src/github.com/rfjakob/eme (from $GOPATH)
internal/cryptocore/hkdf.go:7:2: cannot find package "golang.org/x/crypto/hkdf" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/hkdf (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/hkdf (from $GOPATH)
internal/configfile/scrypt.go:8:2: cannot find package "golang.org/x/crypto/scrypt" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/scrypt (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/scrypt (from $GOPATH)
internal/stupidgcm/prefer.go:6:2: cannot find package "golang.org/x/sys/cpu" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/cpu (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/cpu (from $GOPATH)
internal/syscallcompat/eintr.go:6:2: cannot find package "golang.org/x/sys/unix" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/unix (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/unix (from $GOPATH)
Compiling for armeabi-v7a...
internal/siv_aead/siv_aead.go:9:2: cannot find package "github.com/jacobsa/crypto/siv" in any of:
        /home/dhammel/Apps/go/src/github.com/jacobsa/crypto/siv (from $GOROOT)
        /home/dhammel/go/src/github.com/jacobsa/crypto/siv (from $GOPATH)
internal/cryptocore/cryptocore.go:13:2: cannot find package "github.com/rfjakob/eme" in any of:
        /home/dhammel/Apps/go/src/github.com/rfjakob/eme (from $GOROOT)
        /home/dhammel/go/src/github.com/rfjakob/eme (from $GOPATH)
internal/cryptocore/hkdf.go:7:2: cannot find package "golang.org/x/crypto/hkdf" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/hkdf (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/hkdf (from $GOPATH)
internal/configfile/scrypt.go:8:2: cannot find package "golang.org/x/crypto/scrypt" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/crypto/scrypt (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/crypto/scrypt (from $GOPATH)
internal/stupidgcm/prefer.go:6:2: cannot find package "golang.org/x/sys/cpu" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/cpu (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/cpu (from $GOPATH)
internal/syscallcompat/eintr.go:6:2: cannot find package "golang.org/x/sys/unix" in any of:
        /home/dhammel/Apps/go/src/golang.org/x/sys/unix (from $GOROOT)
        /home/dhammel/go/src/golang.org/x/sys/unix (from $GOPATH)
Done.

my go modules are stored in ~/go/pkg. There is nothing stored in ~/go/src, not even after running the go get command. So, I'm not sure why it seems its checking in ~/go/src

**Comment by hardcore-sushi** [comment]
That seems better. Try to re-run the `go get` command specified in README but replace `get` with `install`.

**Comment by dhammel** [comment]
Well it's successfully built now, actually all I needed to do was re-run the go get command, didn't need to use the install method. I had deleted the contents from ~/go/pk, and just re-ran the go get command, and they were downloaded into ~/go/src. Afterwards the build script completed without errors, and I managed to build a signed apk through android studio's gui.

Thank you for your help with this.
